drivers/xen/gnttab.c

```diff
@@ -38,6 +40,9 @@ LOG_MODULE_REGISTER(xen_gnttab);
 #define NR_GRANT_ENTRIES \
 	(NR_GRANT_FRAMES * XEN_PAGE_SIZE / sizeof(grant_entry_v1_t))
 
+BUILD_ASSERT(2 * DT_REG_SIZE_BY_IDX(DT_INST(0, xen_xen), 0) <= CONFIG_KERNEL_VM_SIZE);
```

Comment:
It would be nice to have comment describing why are you multiplying region size by two.

Reply:
I've performed testing and board always has been crashed until the VM size has been increased by 2 times
actually, 30M is enough, but previously I thought that should be enough to have 16M (grant tables) + 8M (old config)